### PR TITLE
Pistons face the proper direction when placed.

### PR DESCRIPTION
**Pull request by aramperes**
Pistons will face the proper direction in regards of the player's location proportional to the block. This attempts to reproduce Vanilla behavior of piston placement.

_This has nothing to do with the functionality of pistons, yet...!_

### CODE REVIEW

File: src/main/java/net/glowstone/block/blocktype/BlockPiston.java [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+            if (offset - (double) bLoc.getBlockY() > 2.0)
+                return BlockFace.UP;
+
+            if ((double) bLoc.getBlockY() - offset > 0.0)
+                return BlockFace.DOWN;
+        }
+        return player.getDirection().getOppositeFace();
+    }
+
+    private int rawFace(BlockFace face) {
```
Comment (mastercoms) [inline src/main/java/net/glowstone/block/blocktype/BlockPiston.java]: Could you rename this to `getRawFace`?

Comment (mastercoms) [inline src/main/java/net/glowstone/block/blocktype/BlockPiston.java]: Could you rename this to `getRawFace`?

**Comment by MOMOTHEREAL** [comment]
*Why is my git setup so messed up* (._.)
Should be good to go.

**Comment by mastercoms** [comment]
Make a default constructor with no args for non-sticky pistons.

**Comment by mastercoms** [comment]
I updated Glowstone++ to fix compatibility issues with plugins.

**Comment by MOMOTHEREAL** [comment]
Should be fine now. I will add more features to pistons when I get the time to do so.

File: src/main/java/net/glowstone/block/blocktype/BlockPiston.java [hunk 2]
```diff
@@ -0,0 +1,71 @@
...
+
+    public BlockPiston() {
+        sticky = false;
+        setDrops(new ItemStack(Material.PISTON_BASE));
+    }
+
+    public BlockPiston(boolean sticky) {
+        this.sticky = sticky;
+
+        if (!sticky) {
```
Comment (mastercoms) [inline src/main/java/net/glowstone/block/blocktype/BlockPiston.java]: Flip the if statement, please.

**Comment by mastercoms** [comment]
Thank you for your contribution! Works perfectly.

**Pull request merged**